### DISCUSSION TITLE
How to set focus and cursor on editor if document is not empty?

### DISCUSSION THREAD

**Discussion by isaacadariku** [Flutter-Bounty-Hunters/super_editor#883] [Q&A] [answered]
I have a problem with focusing on the editor on the Web and MacOS. If the document is empty, the focus is set on the editor and I can type the text as soon as it opens. But if the document is empty, the focus is not set on the editor and I can't type text. I tried to use the focus() method, but it still didn't work. Please how can I set focus on the editor and also make the cursor show up as soon the editor is open?

You can also use the super editor example to test it out: https://editor.superlist.com/

Thanks!

**Comment by matthew-carroll** [comment]
Can you provide a minimal code sample that shows how you're trying to use the editor?

**Comment by isaacadariku** [comment]
@matthew-carroll thanks for the response, please find the reproduction below.

## Reproduction:
1. On the example app follow this path `super_editor/example/lib/demos/example_editor/example_editor.dart`
- Add `autofocus` property to the SuperEditor widget to `true` (line 392)
![Screenshot 2023-01-23 at 17 51 18](https://user-images.githubusercontent.com/46024202/214099782-cbf9f549-daad-43cc-aec9-38881ac5b096.png)


2. Launch the Mac or Web desktop version of the example app
3. Notice that the blinking cursor is not visible and the autofocus is not working
4. Notice this error below appears in the console:
```
flutter: Initializing logger: ExampleApp

════════ Exception caught by foundation library ════════════════════════════════
The following NoSuchMethodError was thrown while dispatching notifications for FocusNode:
The method 'findLastSelectablePosition' was called on null.
Receiver: null
Tried calling: findLastSelectablePosition()

When the exception was thrown, this was the stack
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:38:5)
#1      _EditorSelectionAndFocusPolicyState._onFocusChange
package:super_editor/…/default_editor/document_focus_and_selection_policies.dart:110
#2      ChangeNotifier.notifyListeners
package:flutter/…/foundation/change_notifier.dart:351
#3      FocusNode._notify
package:flutter/…/widgets/focus_manager.dart:1038
#4      FocusManager._applyFocusChange
package:flutter/…/widgets/focus_manager.dart:1804
(elided 2 frames from dart:async)
The FocusNode sending notification was: FocusNode#2c45c([PRIMARY FOCUS])
    context: Focus
    PRIMARY FOCUS
════════════════════════════════════════════════════════════════════════════════
```
## Expected result:
The editor should accept keyboard input automatically and the blinking cursor should be visible on all platforms when the `autofocus` property is set to `true`.
## Actual result:
The editor does not accept keyboard input automatically and the blinking cursor is not visible and the `autofocus` is not working on Mac desktop and web platforms when the autofocus property is set to `true`.
## Additional information:
The issue is not reproducible on the iOS/Android versions of the example app.
The latest version of the example app was used for the test.

**Comment by angelosilvestre** [comment, marked as answer]
@isaacadariku This issue was fixed in https://github.com/superlistapp/super_editor/pull/1013

**Reply by isaacadariku** [reply]
Thanks @angelosilvestre it works as expected. I really appreciate 🙏🏽
